pystac/catalog.py

```diff
@@ -230,7 +230,7 @@ def add_child(
         self,
         child: Union["Catalog", "Collection_Type"],
         title: Optional[str] = None,
-        strategy: Optional[HrefLayoutStrategy] = None,
+        strategy: HrefLayoutStrategy = BestPracticesLayoutStrategy(),
```

Comment:
Yeah, constructing as the default value is generally dangerous -- it leaves us open to shared-mutable-object concerns. We probably want to keep this as `None` default, and just document the behavior better.

Reply:
Erf. Yeah. Thanks. Will update.